### ISSUE TITLE
Encounter error when build on MacOS

### ISSUE THREAD

**Issue by ariesdevil** [YugaByte/yugabyte-db#59]
```shell
--------------------------------------------------------------------------------
Building uninstrumented dependencies
--------------------------------------------------------------------------------

Have to rebuild aws-sdk-cpp (uninstrumented):
Old build stamp for aws-sdk-cpp: None (from /Users/ariesdevil/Dropbox/workspace/cc_workspace/yugabyte-db/thirdparty/build/uninstrumented/.build-stamp-aws-sdk-cpp)
New build stamp for aws-sdk-cpp: git_commit_sha1=2150bee06359797430fefbc2162678e1cce7ce00
git_diff_sha256=e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855


--------------------------------------------------------------------------------
Building aws-sdk-cpp (uninstrumented)
--------------------------------------------------------------------------------
aws-sdk-cpp (uninstrumented) -- Found Git: /usr/local/bin/git (found version "2.12.2")
aws-sdk-cpp (uninstrumented) CMake Warning (dev) at cmake/resolve_platform.cmake:7 (if):
aws-sdk-cpp (uninstrumented)   Policy CMP0054 is not set: Only interpret if() arguments as variables or
aws-sdk-cpp (uninstrumented)   keywords when unquoted.  Run "cmake --help-policy CMP0054" for policy
aws-sdk-cpp (uninstrumented)   details.  Use the cmake_policy command to set the policy and suppress this
aws-sdk-cpp (uninstrumented)   warning.
aws-sdk-cpp (uninstrumented)
aws-sdk-cpp (uninstrumented)   Quoted variables like "APPLE" will no longer be dereferenced when the
aws-sdk-cpp (uninstrumented)   policy is set to NEW.  Since the policy is not set the OLD behavior will be
aws-sdk-cpp (uninstrumented)   used.
aws-sdk-cpp (uninstrumented) Call Stack (most recent call first):
aws-sdk-cpp (uninstrumented)   CMakeLists.txt:80 (include)
aws-sdk-cpp (uninstrumented) This warning is for project developers.  Use -Wno-dev to suppress it.
aws-sdk-cpp (uninstrumented)
aws-sdk-cpp (uninstrumented) -- TARGET_ARCH not specified; inferring host OS to be platform compilation target
aws-sdk-cpp (uninstrumented) -- Building AWS libraries as static objects
aws-sdk-cpp (uninstrumented) -- Building project version: v0.9.5.0
aws-sdk-cpp (uninstrumented) CMake Error at CMakeLists.txt:107 (project):
aws-sdk-cpp (uninstrumented)   VERSION "v0.9.5.0" format invalid.
aws-sdk-cpp (uninstrumented)
aws-sdk-cpp (uninstrumented)
aws-sdk-cpp (uninstrumented) -- Configuring incomplete, errors occurred!
```

**Comment by mbautin** [comment]
@ariesdevil: thank you for reporting this issue! I think this is caused by the AWS SDK build picking up YugaByte DB version number as its own version, because we build it from a tarball extracted inside of a subdirectory of the YugaByte DB repository. Instead of checking out the latest release tag of YugaByte DB, try checking out `master`, or its commit (currently `ab998f81cbae8c82b6efcafe8c168ee56eb203c7`), or the commit corresponding to `v0.9.5.0` (655c4f7b12a0affebc8601c4e00d84c27ec16136). If none of the above helps, please share a few details about your environment to help us diagnose and fix the issue faster:
```
sw_vers
brew --version
brew list --versions
git --version
```

**Comment by ariesdevil** [comment]
@mbautin Try `ab998f81cbae8c82b6efcafe8c168ee56eb203c7` but same issue.

My env:
```shell
sw_vers
ProductName:	Mac OS X
ProductVersion:	10.13.3
BuildVersion:	17D102

brew --version
Homebrew 1.4.3
Homebrew/homebrew-core (git revision 0ba41; last commit 2018-01-15)

brew list --versions
autoconf 2.69
automake 1.15.1
bash 4.4.12
bison 3.0.4_1 3.0.4
boost 1.65.1
ccache 3.3.4_1
cmake 3.7.2 3.9.4 3.8.0
coreutils 8.28_1
flex 2.6.3 2.6.4
gcc 7.1.0 7.2.0_1 7.2.0
gcc@4.9 4.9.3
gd 2.2.4_1
gdb 7.12.1 8.0.1
gdbm 1.13 1.14.1_1
gettext 0.19.8.1
gflags 2.2.1
ghostscript 9.19 9.22
gifsicle 1.91
git 2.12.2
git-extras 4.2.0
git-lfs 1.5.3
glib 2.54.1
gnu-tar 1.30
gnupg 2.0.30_2
gnutls 3.5.15
libtool 2.4.6_1
pkg-config 0.29.2
protobuf 3.4.1 3.2.0_1 3.1.0
protobuf-c 1.3.0_1 1.2.1_2
protobuf@2.5 2.5.0
pstree 2.39
wget 1.18
zlib 1.2.11
zstd 1.3.1

git --version
git version 2.12.2
```

**Comment by mbautin** [comment]
@ariesdevil: got it, thanks for the info! Can you try one more thing? Just remove the `.git` directory after doing the checkout to prevent AWS SDK build script from using YugaByte DB's git branch as its own version. A proper fix is coming.

**Comment by mbautin** [comment]
Actually, the above does not work. Will post another workaround.

**Comment by mbautin** [comment]
@ariesdevil: fixed this by the latest commit (ad99b068a3a5172e7befe9210af8c346d3105ac9). Please give it a try and let us know if there is still an issue!

**Comment by ariesdevil** [comment]
@mbautin It works now. Thx
